src/particle_exchanger.h

```diff
@@ -206,7 +201,21 @@ void ParticleExchanger_t<Halo_T>::QueryParticles()
   for (auto &&p : RoamParticles)
   {
     if (p.Id != SpecialConst::NullParticleId)
+    {
       p = snap.Particles[p.Id]; // query the particle property; may need to spawn particles due to star formation here.
+    }
+    else
+    {
+#ifdef DM_ONLY
+      /* All DM particles should be found */
+      assert(false); // Assume p.Type=TypeDM since type is not stored explicitly
+#else
+      /* In hydro runs DM and star particles are not expected to disappear.
```

Comment:
Although this is true for our current simulations, those using higher resolution ones (or annihilating dark matter) might have this issue. We should mention this possibility as a warning for potential future users.

Reply:
I agree about annihilating DM but why would higher resolution simulations be a problem?

Reply:
If you resolve individual massive stars (e.g. LYRA) then if one of them goes supernovae it will not be a particle. It would be implementation dependent, of course, but I could see this being an issue for high res simulations of dwarfs, for example. This is just me thinking about potential future problems

Reply:
Should I change it to check that tracers are always found? I think we assume that's the case in other parts of the code so I wouldn't be adding any extra assumptions then.

Reply:
I've changed it to require Type=TypeMax or !IsTracer() for particles which can't be found. I think that's better than hard coding assumptions about what the tracers are.